### ISSUE TITLE
Check for updates button doesn't check for updates

### ISSUE THREAD

**Issue by peppy** [ppy/osu#12324]
Seems to do nothing.

**Comment by frenzibyte** [comment]
Looks to work fine on my end. Both on macOS and Windows with Squirrel, even with removing the automatic update checking at `UpdateManager.LoadComplete()`.

**Comment by peppy** [comment]
I can still reproduce it not checking at my end:

https://user-images.githubusercontent.com/191335/114432642-e1a95680-9bfb-11eb-894f-8f94dfe38733.mp4

Haven't had a chance to look into it further yet.

**Comment by peppy** [comment]
I've further tested this on macOS and can't reproduce the issue. It's likely some kind of web request caching logic within Squirrel if anything.

For what it's worth, this came up when i had osu! already running and pushed a new release. Pressing the "Check for updates" button had no effect, but restarting the game found the update immediately.

**Comment by Joehuu** [comment]
I never pressed the button until I realized there was no update notification at the start when testing older releases, and I don't think I restarted osu! that much.

Only thing I can think of is that other apps I was using at the time were using the api (e.g. some VSCode extension), and I can remember restarting that an abnormal amount of times.

**Comment by peppy** [comment]
Yeah, quite feasible since it's per IP at that point. The eventual plan would be to not rely on github in the first place for these calls, but we could consider creating an access key for the client to bypass the rate limiting.
